xds/src/test/java/io/grpc/xds/EnvoyServerProtoDataTest.java

```diff
@@ -0,0 +1,134 @@
+import io.envoyproxy.envoy.api.v2.listener.Filter;
+import io.envoyproxy.envoy.api.v2.listener.FilterChain;
+import io.envoyproxy.envoy.api.v2.listener.FilterChainMatch;
+import io.grpc.xds.EnvoyServerProtoData.Listener;
+import io.grpc.xds.internal.sds.CommonTlsContextTestsUtil;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Unit tests for {@link EnvoyServerProtoData}.
+ */
+@RunWith(JUnit4.class)
+public class EnvoyServerProtoDataTest {
+
+  @Test
+  public void listener_convertFromListenerProto() {
+    io.envoyproxy.envoy.api.v2.core.Address address =
+        io.envoyproxy.envoy.api.v2.core.Address.newBuilder()
+            .setSocketAddress(SocketAddress.newBuilder()
+                .setPortValue(8000)
+                .setAddress("10.2.1.34")
+                .build())
+            .build();
+    io.envoyproxy.envoy.api.v2.Listener listener =
+        io.envoyproxy.envoy.api.v2.Listener.newBuilder()
+            .setName("8000")
+            .setAddress(address)
+            .addFilterChains(createOutFilter())
+            .addFilterChains(createInFilter())
+            .build();
+
+    Listener xdsListener = Listener.fromEnvoyProtoListener(listener);
+    assertThat(xdsListener.getName()).isEqualTo("8000");
+    assertThat(xdsListener.getAddress()).isEqualTo("10.2.1.34:8000");
+    List<EnvoyServerProtoData.FilterChain> filterChains = xdsListener.getFilterChains();
+    assertThat(filterChains).isNotNull();
+    assertThat(filterChains.size()).isEqualTo(2);
+    EnvoyServerProtoData.FilterChain outFilter = filterChains.get(0);
+    assertThat(outFilter).isNotNull();
+    EnvoyServerProtoData.FilterChainMatch outFilterChainMatch = outFilter.getFilterChainMatch();
+    assertThat(outFilterChainMatch).isNotNull();
+    assertThat(outFilterChainMatch.getDestinationPort()).isEqualTo(8000);
+    assertThat(outFilterChainMatch.getApplicationProtocols()).isEmpty();
+    assertThat(outFilterChainMatch.getPrefixRanges()).isEmpty();
+    assertThat(outFilter.getDownstreamTlsContext())
+        .isEqualTo(DownstreamTlsContext.getDefaultInstance());
+
+    EnvoyServerProtoData.FilterChain inFilter = filterChains.get(1);
+    assertThat(inFilter).isNotNull();
+    EnvoyServerProtoData.FilterChainMatch inFilterChainMatch = inFilter.getFilterChainMatch();
+    assertThat(inFilterChainMatch).isNotNull();
+    assertThat(inFilterChainMatch.getDestinationPort()).isEqualTo(8000);
+    assertThat(inFilterChainMatch.getApplicationProtocols()).containsExactly("managed-mtls");
+    assertThat(inFilterChainMatch.getPrefixRanges()).containsExactly(
+        new EnvoyServerProtoData.CidrRange("10.20.0.15", 32));
+    DownstreamTlsContext inFilterTlsContext = inFilter.getDownstreamTlsContext();
+    assertThat(inFilterTlsContext).isNotNull();
+    CommonTlsContext commonTlsContext = inFilterTlsContext.getCommonTlsContext();
+    assertThat(commonTlsContext).isNotNull();
```

Comment:
Proto fields will never be null.

Anyway, it's not necessary to write tests for this data conversion class (unless there is some complex logic other than straightly populating data from proto messages).

Reply:
Tests are meant to catch potential regressions in the future

Reply:
i agree the statement, but it doesn't apply to this case. just like we don't unittest standard library. we will probably have a fix from next import.

Reply:
My comment was about "not needing a test" or "test not being useful". I agree that the "NotNull" checks can be (should be?) removed based on linter issues.

Reply:
oops, yes i believe the conversion tests are useful.
other test will verify behavior using converted data class. without proto to converted verification, we can't trust other tests.